code/datums/saymode.dm

```diff
@@ -64,3 +65,26 @@
 		AI.holopad_talk(message, language)
 		return FALSE
 	return TRUE
+
+/datum/saymode/slime_link
+	key = MODE_KEY_SLIMELINK
+	mode = MODE_SLIMELINK
+	early = TRUE
+
+/datum/saymode/slime_link/handle_message(mob/living/user, message, datum/language/_language)
+	. = FALSE
+	if(!user || !user.mind)
+		return TRUE
+	if(!length(message))
+		return TRUE
+	if(ishuman(user))
+		var/mob/living/carbon/human/h_user = user
+		if(isstargazer(h_user))
+			var/datum/species/oozeling/stargazer/stargazer = h_user.dna.species
+			stargazer.slime_chat(h_user, message)
+			return
+	var/datum/weakref/owner_ref = GLOB.slime_links_by_mind[user.mind]
+	var/datum/species/oozeling/stargazer/stargazer = owner_ref?.resolve()
+	if(!stargazer || !istype(stargazer))
+		return TRUE
+	stargazer.slime_chat(user, message)
```

Comment:
```suggestion
	var/datum/species/oozeling/stargazer/slimelink_owner = owner_ref?.resolve()
	if(!slimelink_owner || !istype(slimelink_owner))
		return TRUE
	slimelink_owner.slime_chat(user, message)
```
`stargazer` here feels off to me. I feel like it implies the chat says is the slime link owner, but it's not theirs. line 83 is fine because it's themselves.